src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/LPG2RDFConfigurationImpl.java

```diff
@@ -14,25 +14,21 @@ public class LPG2RDFConfigurationImpl implements LPG2RDFConfiguration {
     protected final String PROPERTY = "property/";
 
     protected final Node label;
+    protected final NodeMapping nodeMapping;
```

Comment:
I think variable NODE is required because it is used in mapsToNode method.

Reply:
Oh, right. I didn't remember we had such a function as well.

Now, this means the new `NodeMapping` interface that you have created for this PR should actually have a third function.

This function may be called `isPossibleResult` and the JavaDoc for it can say something like: "Returns true if the given RDF term is in the image of this node mapping and, thus, may be one of the RDF terms returned by the {@link #map(LPGNode)} function."

Once you have added this function, you can use it to implement the `mapsToNode` method.

Then, you will also need to add an implementation for this new function to each of the two classes that implement the `NodeMapping` interface. The implementation for the IRI-version can be like the current content of the `mapsToNode` method and the implementation for the blank node-version just needs to check that the given RDF term is a blank node.

Reply:
I added the isPossibleResult method.